paddle/fluid/framework/details/execution_strategy.h

```diff
@@ -25,7 +25,10 @@ struct ExecutionStrategy {
   size_t num_threads_{0};
   bool use_cuda_{true};
   bool allow_op_delay_{false};
-  size_t num_iteration_per_drop_scope_{1};
+  // If we set this to 1, we will delete all variables when finish a batch. and
+  // this will loss 15%+ performance.
+  // Please be aware about this parameters.
+  size_t num_iteration_per_drop_scope_{100};
```

Comment:
@chengduoZH What do you think about this change?

Reply:
In some models, this change will make training faster. Because if the local scopes are not dropped, the next iteration will reuse the memory, it can avoid alloc operation again.
But, in [`TransformData`](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/framework/operator.cc#L1054-L1068), the transformed data will be stored in the local scope, and it will be not deleted until the drop_scope is called, for example, Transformer(big mode).
So this option doesn't have a best default value.